src/NLog/Internal/FileAppenders/FileAppenderCache.cs

```diff
@@ -95,6 +111,27 @@ public FileAppenderCache(int size, IFileAppenderFactory appenderFactory, ICreate
         /// </summary>
         public int Size { get; private set; }
 
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__
+        /// <summary>
+        /// If `true`, files will be watched for external file archiving and invalidated. 
+        /// Call the <see cref="FileAppenderCache.InvalidateAppendersForInvalidFiles"/> method to "flush" the list
+        /// of invalidated files.
+        /// </summary>
+        public bool WatchExternalFileArchiving
```

Comment:
Can we move this after `ExternalFileArchivingWatcher_OnChange`, so we can use the same `#if`?